pandas/io/parsers.py

```diff
@@ -1597,12 +1591,8 @@ def _get_name(icol):
                 if i == icol:
                     return c
 
-        to_remove = []
-        index = []
-        for idx in self.index_col:
-            name = _get_name(idx)
-            to_remove.append(name)
-            index.append(data[name])
+        to_remove = [_get_name(idx) for idx in self.index_col]
```

Comment:
Same.

Reply:
Let's revert this as well.  The for-loop is clearer as to what's going on.